src/main/java/cn/nukkit/blockentity/impl/FurnaceBlockEntity.java

```diff
@@ -45,7 +45,16 @@ protected FurnaceBlockEntity(BlockEntityType<?> type, Chunk chunk, Vector3i posi
 
     public FurnaceBlockEntity(BlockEntityType<?> type, Chunk chunk, Vector3i position) {
         super(type, chunk, position);
-        this.inventory = new FurnaceInventory(this, InventoryType.FURNACE);
+        InventoryType invType;
+        Block furnace = getBlock();
+        if (furnace.getId() == BLAST_FURNACE || furnace.getId() == LIT_BLAST_FURNACE) {
```

Comment:
Factory for this too.

Reply:
Didn't end up using factory for this, but made the constructors work properly.